### PR TITLE
Fix #1425, delete document using rm_call on drop index command

### LINKED ISSUE #1425: [MAJOR] 1.99.2 memory consumption off the charts
With redisearch 1.99.2 (M2) there seems to be an issue with memory consumption of redis master:
- there are 0 keys in redis
- there are redis indexes created
- only using aof (dump.rdb only for initial sync of slaves)
85M     appendonly.aof
108K    dump.rdb
- redis master uses 5GB (= maxmemory value, otherwise would even use more), causing OOM errors
- redis slaves use less than 1GB
- Issue happens every few days.
- I fixed the issue by deleting the aof/rdb files & restarting redis master, but that's not a permanent solution (since all redis data is lost)
- we didn't have this issue on 1.99.1 (though it may have been lucky, as we didn't use it for too long, but I think it's new in 1.99.1)
- we don't have this issue with RediSearch 1.6.13 (on Redis 6 too) on a master/slave setup with identical redis.conf files.

<details>
  <summary>redis master info memory</summary>

```
redis-cli -h 127.0.0.1 -p 6379 -a foobar keys '*'
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
(empty array)

redis-cli -h 127.0.0.1 -p 6379 -a foobar info memory
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
# Memory
used_memory:5372102192
used_memory_human:5.00G
used_memory_rss:5330169856
used_memory_rss_human:4.96G
used_memory_peak:5373977752
used_memory_peak_human:5.00G
used_memory_peak_perc:99.97%
used_memory_overhead:537737404
used_memory_startup:831560
used_memory_dataset:4834364788
used_memory_dataset_perc:90.00%
allocator_allocated:5372381928
allocator_active:5450297344
allocator_resident:5535256576
total_system_memory:32933359616
total_system_memory_human:30.67G
used_memory_lua:43008
used_memory_lua_human:42.00K
used_memory_scripts:960
used_memory_scripts_human:960B
number_of_cached_scripts:4
maxmemory:5368709120
maxmemory_human:5.00G
maxmemory_policy:allkeys-lru
allocator_frag_ratio:1.01
allocator_frag_bytes:77915416
allocator_rss_ratio:1.02
allocator_rss_bytes:84959232
rss_overhead_ratio:0.96
rss_overhead_bytes:-205086720
mem_fragmentation_ratio:0.99
mem_fragmentation_bytes:-41870312
mem_not_counted_for_evict:0
mem_replication_backlog:536870912
mem_clients_slaves:33972
mem_clients_normal:0
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
```

</details>

When checking on the slave:
<details>
  <summary>redis slave info memory</summary>

```
redis-cli -s /var/run/redis2/redis2.sock info memory
# Memory
used_memory:865816736
used_memory_human:825.71M
used_memory_rss:1267683328
used_memory_rss_human:1.18G
used_memory_peak:4949439792
used_memory_peak_human:4.61G
used_memory_peak_perc:17.49%
used_memory_overhead:2525334
used_memory_startup:831288
used_memory_dataset:863291402
used_memory_dataset_perc:99.80%
allocator_allocated:866645256
allocator_active:1182322688
allocator_resident:1299005440
total_system_memory:32933359616
total_system_memory_human:30.67G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.36
allocator_frag_bytes:315677432
allocator_rss_ratio:1.10
allocator_rss_bytes:116682752
rss_overhead_ratio:0.98
rss_overhead_bytes:-31322112
mem_fragmentation_ratio:1.46
mem_fragmentation_bytes:401928560
mem_not_counted_for_evict:0
mem_replication_backlog:1048576
mem_clients_slaves:0
mem_clients_normal:645470
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
```

</details>

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/pytest/test_followhashes.py [hunk 1]
```diff
@@ -569,4 +569,17 @@ def testWrongFieldType(env):
...
+def testDocIndexedInTwoIndexes():
+    env = Env(moduleArgs='MAXDOCTABLESIZE 50')
+    env.skipOnCluster()
+    env.expect('FT.CREATE idx1 SCHEMA t TEXT').ok()
+    env.expect('FT.CREATE idx2 SCHEMA t TEXT').ok()
+
+    for i in range(1000):
+        env.expect('HSET', 'doc%d' % i, 't', 'foo').equal(1L)
+
+    env.expect('FT.DROPINDEX idx2 DD').ok()
```
Comment (ashtul) [inline src/pytest/test_followhashes.py:582]: failing
Interesting. doc ids are 50 apart...

Comment (MeirShpilraien) [reply src/pytest/test_followhashes.py:582]: Fixed

**Review by rafie — APPROVED** [review]

File: src/doc_table.h [hunk 2]
```diff
@@ -74,7 +74,7 @@ typedef struct {
   if (md) ++md->ref_count;
 
 #define DOCTABLE_FOREACH(dt, code)                                           \
-  for (size_t i = 1; i < dt->cap; ++i) {                                     \
+  for (size_t i = 0; i < dt->cap; ++i) {                                     \
```
Comment (ashtul) [inline src/doc_table.h:77]: nice catch

**Review by ashtul — APPROVED** [review]
nice...

**Pull request merged**